0000-pre-shared-key-v2.md

```diff
@@ -0,0 +1,291 @@
+8. Sender creates a [Payment Chunk Request](#payment-chunk-request-response) with a random Payment ID, a Sequence number of 0, and a Payment Amount and Chunk Amount set to the destination amount determined in step 7. Sender [encrypts](#encryption-envelope) the request.
+9. Sender generates the payment condition as described [below](#payment-chunk-condition).
+10. Sender sends a payment using the source amount determined in step 7, the condition from step 9, and the encrypted request as the data.
+11. Receiver is notified of the incoming transfer and decrypts the data to determine that it is a payment chunk.
+12. Receiver checks to ensure that the incoming transfer amount is at least the chunk amount specified in the request.
+13. Receiver generates fulfillment as described [below](#payment-chunk-fulfillment) and fulfills the incoming transfer.
+14. Sender is notified of the outgoing transfer being fulfilled.
+
+### Chunked Payment
+
+1. Sender creates a [Payment Chunk Request](#payment-chunk-request-response) with a random Payment ID, a Sequence number of 0 and a Chunk Amount of 0. In the case of a fixed destination amount, sender sets the Payment Amount to the destination amount. In the case of a fixed source amount, sender sets the Payment Amount to 0. Sender [encrypts the request](#encryption-envelope).
+2. Sender generates the payment condition as described [below](#payment-chunk-condition).
+3. Sender sends a payment with the condition from step 2., the encrypted request from step 1. as the data, and an arbitrary source amount.
+4. Receiver is notified of the incoming transfer and decrypts the data to determine it is for a payment chunk.
+5. Receiver creates a [Payment Chunk Response](#payment-chunk-request-response) with the same Payment ID, Sequence, and Payment ID. Receiver sets the Chunk Amount to the amount that arrived on the incoming transfer. Receiver [encrypts the response](#encryption-envelope).
+6. Receiver generates fulfillment as described [below](#payment-chunk-fulfillment) and fulfills the incoming transfer.
+7. Sender is notified of the outgoing transfer being fulfilled and decrypts the attached data.
+8. Sender determines the exchange rate by dividing the Chunk Amount from the receiver by the chunk's source amount.
+9. Sender repeats step 1-8 until the fixed source or destination amount has been sent or delivered, respectively. Sender MAY adjust the chunk amount upwards or downwards in response to successful and unsuccessful payments.
+
+## Specification
+
+### Encryption
+
+Request and response data are encrypted using [AES-GCM](https://en.wikipedia.org/wiki/Galois/Counter_Mode) with a 16-byte Initialization Vector (IV).
+
+#### Encryption Envelope
+
```

Comment:
Strongly recommend using the format from RFC8188 rather than making up our own.
See: http://httpwg.org/specs/rfc8188.html#header

This would make PKS over ILP over HTTP compliant with RFC8188.

Reply:
> This would make PKS over ILP over HTTP compliant with RFC8188.

This wouldn't be true if we [bring back the ILP packet](https://github.com/interledger/rfcs/pull/347#issuecomment-350759502), because the HTTP body will be the ILP packet and this envelope will be inside there. I think it's more important to be consistent with the rest of our stack (using OER)

Reply:
I agree. This makes me sad 😢 

1. Put ILP Address in HTTP headers use RFC 8188 as is. Profit
OR
2. Put ILP Address in packet in body. Invent everything from scratch.